chunk/chunk_store.go

```diff
@@ -130,18 +133,26 @@ func NewStore(cfg StoreConfig) (*Store, error) {
 	}, nil
 }
 
-func chunkName(userID, chunkID string) string {
-	return fmt.Sprintf("%s/%s", userID, chunkID)
-}
-
 // Put implements ChunkStore
 func (c *Store) Put(ctx context.Context, chunks []Chunk) error {
 	userID, err := user.Extract(ctx)
 	if err != nil {
 		return err
 	}
 
-	err = c.putChunks(ctx, userID, chunks)
+	// Encode the chunk first, as a side effect is to calculate the checksum.
```

Comment:
"as a side effect is to calculate the checksum" is grammatically unusual. Maybe drop the "as"? Or just the whole comment, since I can't see why it matters at this point in the code.